benchmarks/maxtext_xpk_runner.py

```diff
@@ -0,0 +1,412 @@
+      slow_worker_index = returncodes.index(None)
+      slow_worker_text = per_command_name[slow_worker_index]
+      slow_str = (
+          f', task {slow_worker_text} still working, logfile'
+          f' {output_logs[slow_worker_index].name}'
+      )
+    else:
+      slow_str = ''
+    print(
+        f'[t={seconds_elapsed:.2f}, {jobname}] Completed'
+        f' {completed}/{total}{slow_str}'
+    )
+    if max_returncode > 0:
+      failing_index = [
+          i for i, x in enumerate(returncodes) if x is not None and x > 0
+      ][0]
+      print(f'Terminating all {jobname} processes since at least one failed.')
+      print(
+          f'Failure is {per_command_name[failing_index]}'
+          f' and logfile {output_logs[failing_index].name}'
+      )
+      for child in children:
+        child.terminate()
+      break
+
+    if completed == total:
+      break
+
+    time.sleep(1)
+  return max_returncode, returncodes
+
+
+def run_command_with_updates(command, task, verbose=True) -> int:
+  """Generic run commands function with updates.
+
+  Args:
+    command: command to execute
+    task: user-facing name of the task
+    global_args: user provided arguments for running the command.
+    verbose: shows stdout and stderr if set to true. Set to True by default.
+
+  Returns:
+    0 if successful and 1 otherwise.
+  """
+
+  if verbose:
+    print(
+        f'Task: `{task}` is implemented by `{command}`, streaming output live.'
+    )
+    with subprocess.Popen(
+        command,
+        stdout=sys.stdout,
+        stderr=sys.stderr,
+        shell=True,
+    ) as child:
+      i = 0
+      while True:
+        return_code = child.poll()
+        if return_code is None:
+          print(f'Waiting for `{task}`, for {i} seconds')
+          time.sleep(1)
+          i += 1
+        else:
+          print(f'Task: `{task}` terminated with code `{return_code}`')
+          return return_code
+  else:
+    print(
+        f'Task: `{task}` is implemented by `{command}`, hiding output unless'
+        ' there is an error.'
+    )
+    try:
+      subprocess.check_output(command, shell=True, stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+      print(
+          f'Task: `{task}` terminated with ERROR `{e.returncode}`, printing'
+          ' logs'
+      )
+      print('*' * 80)
+      print(e.output)
+      print('*' * 80)
+      return e.returncode
+    print(f'Task: `{task}` succeeded.')
+    return 0
+
+
+def build_user_command(
+    model: maxstar_model_configs.MaxTextModel,
+    num_slices: int,
+    num_steps: int,
+    libtpu_type: LibTpuType,
+    libtpu_date: str,
+    cluster_config: XpkConfig,
+    buffer_size: int,
+):
+  config_tuning_params = ''
+  for key, value in model.tuning_params.items():
+    config_tuning_params += f'{key}={value} '
+
+  install_libtpu_cmd = ''
+  if libtpu_type == LibTpuType.V6E_NIGHTLY_WHEEL:
+    install_libtpu_cmd += f' pip install *${libtpu_date}*.whl &&'
```

Comment:
Is this actually specific to v6e? Can this option be renamed to NIGHTLY_WHEEL if its not?

Reply:
I think we can safely remove it, as now v6e libtpu can use nightly version using jax-release channel. @Obliviour to confirm?